### DISCUSSION TITLE
security onion ver. 2.3.70 installation hangs

### DISCUSSION THREAD

**Discussion by sinakeshmiri** [Security-Onion-Solutions/securityonion#5253] [Unsupported Versions]
installation hangs on Applying elasticsearch Salt state stage (using Ubuntu 18.04-distributed manger node  - default options) 

![Screenshot from 2021-08-23 11-30-44](https://user-images.githubusercontent.com/72389059/130405928-8594e7a7-b6c2-449c-ac85-147c3e842619.png)
![Screenshot from 2021-08-23 11-30-50 (copy)](https://user-images.githubusercontent.com/72389059/130405978-58d30f97-af15-49a8-abce-dfcd405d598d.png)

![Screenshot from 2021-08-23 11-45-54](https://user-images.githubusercontent.com/72389059/130406239-5ce39981-2b29-4358-9332-d103aa203022.png)

Checking Docker status

    Docker ------------------------------------------------------------------------------------------------------------------------ [ OK ]    

Checking container statuses

    so-aptcacherng ---------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-dockerregistry ------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-elasticsearch ------------------------------------------------------------------------------------------------------ [ WAIT_START ]    
    so-idstools ------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-nginx ---------------------------------------------------------------------------------------------------------------------- [ OK ]

**Comment by dougburks** [comment]
It looks like you only have 8GB RAM. Depending on what options you choose in Setup, you may need to increase that. Please review the Hardware Requirements:
https://docs.securityonion.net/en/2.3/hardware.html

**Reply by sinakeshmiri** [reply]
increased my resources i will try to install again , ty so much . ill let you know about the results

**Reply by sinakeshmiri** [reply]
it wasn't because of ram , i just removed old docker containers from last unsuccessful installation and i could pass that stage but now im stuck with some kind of other errors :)
[ERROR   ] Command '/usr/bin/patch' failed with return code: 1
[ERROR   ] stdout: patching file /tmp/__salt.tmp.d9pofov4 (read from /usr/lib/python3/dist-packages/salt/modules/influxdbmod.py)
[ERROR   ] retcode: 1

**Reply by ericzinnikas** [reply]
Seeing the same issue here, node has 32gb RAM so that is likely not the cause. `sudo docker ps` doesn't actually show the so-elasticsearch container running, but I don't see anything obvious failing directly before this. Tips on where to troubleshoot? (rm old docker containers hasn't helped me here either).

Edit: FWIW issue likely related to the fact that I had some half-successfully installations, while doing dev. Clean install works as expected.

**Reply by sinakeshmiri** [reply]
@ericzinnikas
actually it's because of the Elasticsearch container you're looking in a right place , i know its weird but it worked for me :

remove all containers
then
sudo apt-get remove docker-ce-rootless-extras  docker-ce python3-docker
try to create a container from image you pulled from your unsuccessful installation for elastic-search
and then reboot
after all i'm still facing similar issues with so-dockerregistry so-aptcacherng , influxdb and so on

**Reply by sinakeshmiri** [reply]
[ERROR   ] Specified target /opt/so/conf/docker-registry/etc/config.yml is a directory
[ERROR   ] Container 'so-dockerregistry' is already configured as specified. Failed to start container 'so-dockerregistry': 'Error 400: OCI runtime create failed: container_linux.go:367: starting container process caused: process_linux.go:495: container init caused: rootfs_linux.go:60: mounting "/opt/so/conf/docker-registry/etc/config.yml" to rootfs at "/var/lib/docker/overlay2/ea8dde5a018a6c5d1e5f23ed8ca0dab32dc782ecc0ddd095676fbaaedf3f9ece/merged/etc/docker/registry/config.yml" caused: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type'.
[ERROR   ] Container 'so-dockerregistry' is already configured as specified. Failed to start container 'so-dockerregistry': 'Error 400: OCI runtime create failed: container_linux.go:367: starting container process caused: process_linux.go:495: container init caused: rootfs_linux.go:60: mounting "/opt/so/conf/docker-registry/etc/config.yml" to rootfs at "/var/lib/docker/overlay2/ea8dde5a018a6c5d1e5f23ed8ca0dab32dc782ecc0ddd095676fbaaedf3f9ece/merged/etc/docker/registry/config.yml" caused: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type'.
[ERROR   ] Container 'so-dockerregistry' is already configured as specified. Failed to start container 'so-dockerregistry': 'Error 400: OCI runtime create failed: container_linux.go:367: starting container process caused: process_linux.go:495: container init caused: rootfs_linux.go:60: mounting "/opt/so/conf/docker-registry/etc/config.yml" to rootfs at "/var/lib/docker/overlay2/ea8dde5a018a6c5d1e5f23ed8ca0dab32dc782ecc0ddd095676fbaaedf3f9ece/merged/etc/docker/registry/config.yml" caused: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type'.
[ERROR   ] Container 'so-dockerregistry' is already configured as specified. Failed to start container 'so-dockerregistry': 'Error 400: OCI runtime create failed: container_linux.go:367: starting container process caused: process_linux.go:495: container init caused: rootfs_linux.go:60: mounting "/opt/so/conf/docker-registry/etc/config.yml" to rootfs at "/var/lib/docker/overlay2/ea8dde5a018a6c5d1e5f23ed8ca0dab32dc782ecc0ddd095676fbaaedf3f9ece/merged/etc/docker/registry/config.yml" caused: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type'.
[ERROR   ] Container 'so-dockerregistry' is already configured as specified. Failed to start container 'so-dockerregistry': 'Error 400: OCI runtime create failed: container_linux.go:367: starting container process caused: process_linux.go:495: container init caused: rootfs_linux.go:60: mounting "/opt/so/conf/docker-registry/etc/config.yml" to rootfs at "/var/lib/docker/overlay2/ea8dde5a018a6c5d1e5f23ed8ca0dab32dc782ecc0ddd095676fbaaedf3f9ece/merged/etc/docker/registry/config.yml" caused: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type'.
[ERROR   ] Specified target /opt/so/conf/aptcacher-ng/etc/acng.conf is a directory
[ERROR   ] Container 'so-aptcacherng' is already configured as specified. Failed to start container 'so-aptcacherng': 'Error 400: OCI runtime create failed: container_linux.go:367: starting container process caused: process_linux.go:495: container init caused: rootfs_linux.go:60: mounting "/opt/so/conf/aptcacher-ng/etc/acng.conf" to rootfs at "/var/lib/docker/overlay2/bc2b0915b1ad5d80df010d80bf69af13935125bc724d4699f976b233502a10e0/merged/etc/apt-cacher-ng/acng.conf" caused: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type'.
[ERROR   ] Command '/usr/bin/patch' failed with return code: 1
[ERROR   ] stdout: patching file /tmp/__salt.tmp.e6c4nir_ (read from /usr/lib/python3/dist-packages/salt/states/influxdb_continuous_query.py)
[ERROR   ] retcode: 1
[ERROR   ] Command '/usr/bin/patch' failed with return code: 1
[ERROR   ] stdout: patching file /tmp/__salt.tmp.g6j5l0bb (read from /usr/lib/python3/dist-packages/salt/states/influxdb_retention_policy.py)
[ERROR   ] retcode: 1
[ERROR   ] Command '/usr/bin/patch' failed with return code: 1
[ERROR   ] stdout: patching file /tmp/__salt.tmp.gnyqppwf (read from /usr/lib/python3/dist-packages/salt/modules/influxdbmod.py)
[ERROR   ] retcode: 1
[ERROR   ] Specified target /opt/so/conf/docker-registry/etc/config.yml is a directory
[ERROR   ] Container 'so-dockerregistry' is already configured as specified. Failed to start container 'so-dockerregistry': 'Error 400: OCI runtime create failed: container_linux.go:367: starting container process caused: process_linux.go:495: container init caused: rootfs_linux.go:60: mounting "/opt/so/conf/docker-registry/etc/config.yml" to rootfs at "/var/lib/docker/overlay2/ea8dde5a018a6c5d1e5f23ed8ca0dab32dc782ecc0ddd095676fbaaedf3f9ece/merged/etc/docker/registry/config.yml" caused: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type'.
[ERROR   ] Container 'so-dockerregistry' is already configured as specified. Failed to start container 'so-dockerregistry': 'Error 400: OCI runtime create failed: container_linux.go:367: starting container process caused: process_linux.go:495: container init caused: rootfs_linux.go:60: mounting "/opt/so/conf/docker-registry/etc/config.yml" to rootfs at "/var/lib/docker/overlay2/ea8dde5a018a6c5d1e5f23ed8ca0dab32dc782ecc0ddd095676fbaaedf3f9ece/merged/etc/docker/registry/config.yml" caused: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type'.
[ERROR   ] Container 'so-dockerregistry' is already configured as specified. Failed to start container 'so-dockerregistry': 'Error 400: OCI runtime create failed: container_linux.go:367: starting container process caused: process_linux.go:495: container init caused: rootfs_linux.go:60: mounting "/opt/so/conf/docker-registry/etc/config.yml" to rootfs at "/var/lib/docker/overlay2/ea8dde5a018a6c5d1e5f23ed8ca0dab32dc782ecc0ddd095676fbaaedf3f9ece/merged/etc/docker/registry/config.yml" caused: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type'.
[ERROR   ] Container 'so-dockerregistry' is already configured as specified. Failed to start container 'so-dockerregistry': 'Error 400: OCI runtime create failed: container_linux.go:367: starting container process caused: process_linux.go:495: container init caused: rootfs_linux.go:60: mounting "/opt/so/conf/docker-registry/etc/config.yml" to rootfs at "/var/lib/docker/overlay2/ea8dde5a018a6c5d1e5f23ed8ca0dab32dc782ecc0ddd095676fbaaedf3f9ece/merged/etc/docker/registry/config.yml" caused: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type'.
[ERROR   ] Container 'so-dockerregistry' is already configured as specified. Failed to start container 'so-dockerregistry': 'Error 400: OCI runtime create failed: container_linux.go:367: starting container process caused: process_linux.go:495: container init caused: rootfs_linux.go:60: mounting "/opt/so/conf/docker-registry/etc/config.yml" to rootfs at "/var/lib/docker/overlay2/ea8dde5a018a6c5d1e5f23ed8ca0dab32dc782ecc0ddd095676fbaaedf3f9ece/merged/etc/docker/registry/config.yml" caused: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type'.
[ERROR   ] Command '/usr/bin/patch' failed with return code: 1
[ERROR   ] stdout: patching file /tmp/__salt.tmp.n2j4cem4 (read from /usr/lib/python3/dist-packages/salt/states/influxdb_continuous_query.py)
[ERROR   ] retcode: 1
[ERROR   ] Command '/usr/bin/patch' failed with return code: 1
[ERROR   ] stdout: patching file /tmp/__salt.tmp.2t9glc6y (read from /usr/lib/python3/dist-packages/salt/states/influxdb_retention_policy.py)
[ERROR   ] retcode: 1
[ERROR   ] Command '/usr/bin/patch' failed with return code: 1
[ERROR   ] stdout: patching file /tmp/__salt.tmp.ortsbula (read from /usr/lib/python3/dist-packages/salt/modules/influxdbmod.py)
[ERROR   ] retcode: 1
[ERROR   ] Specified target /opt/so/conf/aptcacher-ng/etc/acng.conf is a directory
[ERROR   ] Container 'so-aptcacherng' is already configured as specified. Failed to start container 'so-aptcacherng': 'Error 400: OCI runtime create failed: container_linux.go:367: starting container process caused: process_linux.go:495: container init caused: rootfs_linux.go:60: mounting "/opt/so/conf/aptcacher-ng/etc/acng.conf" to rootfs at "/var/lib/docker/overlay2/bc2b0915b1ad5d80df010d80bf69af13935125bc724d4699f976b233502a10e0/merged/etc/apt-cacher-ng/acng.conf" caused: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type'.

**Comment by dougburks** [comment]
@sinakeshmiri Please try a fresh OS installation to make sure there is nothing conflicting from previous attempts. If all else fails, please try our Security Onion ISO image.

@ericzinnikas To avoid confusion, please start a separate discussion as mentioned in https://github.com/Security-Onion-Solutions/securityonion/discussions/1720. Thanks!
